File: 2924-find-champion-ii/2924-find-champion-ii.py
class Solution:
    def findChampion(self, n: int, edges: List[List[int]]) -> int:
        in_degree = [0]*n
        
        
        for i in range(len(edges)):
            in_degree[edges[i][1]] += 1
            
        champ = -1
        champ_count = 0
        
        for i in range(len(in_degree)):
            if in_degree[i] == 0:
                champ_count += 1
                champ = i
        
        return champ if champ_count == 1 else -1
            

File: 2924-find-champion-ii/2924-find-champion-ii.c
int findChampion(int n, int** edges, int edgesSize, int* edgesColSize) {
        int in_degree[n];
        for(int i = 0; i<n; i++)
            in_degree[i] = 0;
    
        for(int i = 0; i<edgesSize; i++){
            in_degree[edges[i][1]] += 1;
        }
            
        int champ = -1;
        int champ_count = 0;
        
        for(int i = 0; i<n; i++){
            if(in_degree[i] == 0){
                champ_count++;
                champ = i;
            }
        }
            
        
        return champ_count > 1 ? -1 : champ;
            
}